internal/nostr/nostr.go

```diff
@@ -128,31 +128,32 @@ func NewService(ctx context.Context) (*Service, error) {
 
 	var openSubscriptions []Subscription
 	if err := svc.db.Where("open = ?", true).Find(&openSubscriptions).Error; err != nil {
-		logger.Errorf("Failed to query open subscriptions: %v", err)
+		logger.Errorf("failed to query open subscriptions: %v", err)
 		return nil, err
 	}
 
 	for _, sub := range openSubscriptions {
-		go func(sub Subscription) {
-			ctx, cancel := context.WithCancel(svc.Ctx)
-			svc.mu.Lock()
-			svc.subscriptions[sub.ID] = cancel
-			svc.mu.Unlock()
-			errorChan := make(chan error)
-			go svc.handleSubscription(ctx, &sub, errorChan)
-
-			err := <-errorChan
-			if err != nil {
-				svc.stopSubscription(&sub)
-				svc.Logger.Errorf("error opening subscription %d: %v", sub.ID, err)
-			}
-			svc.Logger.Infof("opened subscription %d", sub.ID)
-		}(sub)
+		go svc.startOpenSubscription(sub)
```

Comment:
It's quite confusing to call `startOpenSubscription` which then calls `handleSubscription` which calls `startSubscription`. Do we need this concept of "open subscriptions"?

Reply:
Changed, should be good now.

`NewService`/`HandleSubscriptions` call `startSubscription` which calls `processEvents` after opening the subscription